porting/UpdatePortsFor2004.rst

```diff
@@ -0,0 +1,64 @@
+.. _UpdatePortsFor2004:
+
+Update ports for Ubuntu Touch 20.04
+===================================
+
+Ubuntu Touch 20.04 introduces a lot of changes underneath, which requires changes from the porters. This document is intended to serves as an overview on how to make your port runs on Ubuntu Touch 20.04. However, every port is different, and may requires more than what is outlined in this document. We'll be available in the UBports porting Telegram group to answer any question that might occur during process.
+
+For GitLab CI-based ports: switch to shared building script & GitLab CI configuration
+-------------------------------------------------------------------------------------
+
+Because of changes in how Ubuntu Touch 20.04's rootfs is made, we need to changes the way device tarballs are built. Instead of cherry-picking script changes from other port's repository, we recommend transitioning to the shared building script. By using the shared script, your port will receive the latest changes in device tarball building as rootfs continue to evolves (e.g. for 22.04 or newer). To transition to the shared script, remove the whole ``build/`` directory, and replace the content of ``build.sh`` with:
+
+.. code-block:: bash
+
+    #!/bin/bash
+    set -xe
+
+    [ -d build ] || git clone https://gitlab.com/ubports/community-ports/halium-generic-adaptation-build-tools -b halium-11 build
```

Comment:
-b halium-11 can be removed, as it is current `main` branch. `main` works for halium-11, but doesn't yet have the compiler selection to use correct toolchain for -10. Otherwise there is no differences.

Reply:
Done.